src/List/NonEmpty.elm

```diff
@@ -175,18 +175,127 @@ product =
 
 
 
---
--- append
--- concat
--- concatMap
--- intersperse
--- map2
--- andMap
---
--- sort
--- sortBy
--- sortWith
--- isSingleton
+-- append ("1", ["2, 3"]) ("4", ["5"])
+
+
+append : NonEmptyList a -> NonEmptyList a -> NonEmptyList a
+append ( h1, t1 ) ( h2, t2 ) =
+    case h2 :: t2 of
+        [] ->
+            ( h1, t1 )
+
+        _ ->
+            foldr cons ( h2, t2 ) ( h1, t1 )
```

Comment:
It seems that thre is no need for deconstruction.

```suggestion
append ne1 ne2 =
    case toList ne2 of
        [] ->
            ne

        _ ->
            foldr cons ( h2, t2 ) ne
```

Reply:
I was thinking about this and this is what I would expect as an implementation of this function:

```elm
append : NonEmptyList a -> NonEmptyList a -> NonEmptyList a
append ( h1, t1 ) ne2 =
    ( h1, t1 ++ toList ne2 )
```

I guess this implementation might be better in terms of performance?